homeassistant/components/google_assistant/__init__.py

```diff
@@ -61,8 +59,6 @@ def _check_report_state(data):
 
 
 GOOGLE_ASSISTANT_SCHEMA = vol.All(
-    cv.deprecated(CONF_ALLOW_UNLOCK, invalidation_version="0.95"),
-    cv.deprecated(CONF_API_KEY, invalidation_version="0.105"),
```

Comment:
Removing this might cause errors if they still have them configured. Instead, we can still accept them, throw a deprecation warning, but remove the rest of the logic.

Reply:
We discussed shortly on discord and I tested a few things. If I set these options in the yaml config and leave these `cv.deprecated` lines but remove the remainder (and adjust these to no longer use the constants), HA will start just fine with but the Google Assistant component will fail to setup and throw an error in the logs & persistent notification. 

HA has been throwing deprecation warnings for 13 or 23 versions for these changes.... I'd say that just a very, very few users should be affected if they still failed to remove these items.

Let me know how you want to proceed and I'll adjust.

Reply:
For now just drop the invalidation logic, and remove the code code that uses them.

You might be able to use vol.Remove() on the keys instead of vol.Optional() to explicitly make sure the data doesn't remain in the validated config.

Reply:
That works. No more setup fail, yet code is cleaned. Tx @elupus!